Buildsystem update

dbatyai opened the pull request.

galpeter commented on tools/scripts/util.py:
```diff
@@ -0,0 +1,73 @@
+        self.cmd = cmd
+
+    def __str__(self):
+        return "Command \"%s\" failed with exit code: %d" % (" ".join(self.cmd), self.code)
```
No need to escape the `"` string, you could simply use: `'..."%s" ...'`

galpeter commented on tools/scripts/travis.sh:
```diff
@@ -0,0 +1,43 @@
+    fi
+
+    command="python tools/scripts/$CMD.py $type--backend=$BACKEND"
+    $command
```
Why do we need this variable?

dbatyai replied:
We can probably get by without it.

galpeter commented on tools/scripts/run-tests.py:
```diff
@@ -28,22 +28,44 @@
+    result.append(("Cppcheck", run_cppcheck(throw=False)))
+
+    print("\n")
+    print("Running unit-tests.")
```
An follow-up would be to move these prints into the run_* methods, it does not make sense to have it outside imho.

galpeter commented on tools/scripts/cppcheck.py:
```diff
@@ -49,31 +50,23 @@ def run_cppcheck():
-    print_result(ret)
-    sys.exit(ret)
+    util.print_success()
+    sys.exit(0)
```
There is no reason to have sys.exit(0) by default 0 is the exit value.

galpeter commented on tools/scripts/build.py:
```diff
@@ -71,85 +72,89 @@ def get_args():
-    print_result(ret)
-    sys.exit(ret)
+    print_success()
+    sys.exit(0)
```
There is no reason to have sys.exit(0) by default 0 is the exit value.

galpeter commented on tools/scripts/build.py:
```diff
@@ -71,85 +72,89 @@ def get_args():
-def build_unit(arguments):
-    build_path = get_build_path(arguments)
+# Check if build is configured
+def check_configured(arguments):
```
A better way to comment python functions is to use the following format:
```python
def method():
    """ help message """
    ....
```

This way the `help` method in python can query the help string.

dbatyai commented on tools/scripts/travis.sh:
```diff
@@ -0,0 +1,43 @@
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+# The views and conclusions contained in the software and documentation are those
```
Yes, it was already verified that we don't need this. Luckily I've copy-pasted the license from the only other script that still contained it.

dbatyai commented on tools/scripts/run-tests.py:
```diff
@@ -28,22 +28,44 @@
+        else:
+            msg = "\033[1;32mPASS\033[0m"
+
+        print("".join([name, ':', ' ' * (longest - len(name) + 1), msg]))
```
I kind of like how the current output looks, but I agree, code-wise it would definitely look better the other way around.

kkristof commented:
LGTM

dbatyai commented:
@galpeter @elecro, I fixed the things you mentioned (I have no idea why they didn't disappear), and I also added a few commit with additional changes on top. Please take a look again.

szledan commented on tools/scripts/build.py:
```diff
@@ -28,128 +28,137 @@
 
-# Create cmake option list from parsed arguments
 def create_options(arguments):
+    """ Create a CMake option list from parsed arguments """
```
Could you put a full stop the end of helps/comments?
